src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/servicebus/QueueClientSupplier.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.servicebus;
+
+import com.microsoft.azure.servicebus.IQueueClient;
+import com.microsoft.azure.servicebus.QueueClient;
+import com.microsoft.azure.servicebus.ReceiveMode;
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.ConnectionException;
+
+import java.util.function.Supplier;
+
+@Component
+public class QueueClientSupplier implements Supplier<IQueueClient> {
+
+    private final String connectionString;
+
+    public QueueClientSupplier(@Value("${servicebus.queue_envelope_send}") String connectionString) {
+        this.connectionString = connectionString;
+    }
+
+    @Override
+    public IQueueClient get() {
+        try {
+            return new QueueClient(
+                new ConnectionStringBuilder(connectionString),
+                ReceiveMode.PEEKLOCK
+            );
+        } catch (InterruptedException | ServiceBusException exception) {
+            Thread.currentThread().interrupt();
```

Comment:
Should only be interrupted when `InterruptedException` is caught

Reply:
I am afraid this will have to stay as it is because otherwise it fails code coverage and another reviewer will block the merge. More than writing code I am trying to make you all happy guys ;-)

Reply:
That's a good point